sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1143,4 +1143,10 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val primitiveUDF = udf((i: Int) => i * 2)
     checkAnswer(df.select(primitiveUDF($"age")), Row(44) :: Row(null) :: Nil)
   }
+
+  test("SPARK-12102: Ignore nullablity when comparing two sides of case") {
+    val df = sqlContext.sql(
```

Comment:
we should put tests in `catalyst` if possible, as they don't need to launch a spark job and can run faster.